woocommerce/woocommerce-gutenberg-products-block
# Use WooCommerce thumbnail images in Blocks

Fixes #2719.
Fixes #2728.

This PR updates several blocks (Cart, Checkout and all Reviews blocks), so they use the WooCommerce thumbnail instead of the full-size image. It's smaller, so it should be more performant, and it honors the cropping settings set in the customizer.

It also adds an _Image sizing_ attribute to the Product image atomic block. So All Products and, in the future, Product block users will be able to choose between the full size image and the cropped one.

cc @LevinMedia I added this copy to the new attribute, let me know if it makes sense or we should use something else:
<img src="https://user-images.githubusercontent.com/3616980/85137101-f4076d80-b240-11ea-9e4f-a96fd9ddeb42.png" alt="Screenshot" width="274" />

### Screenshots

_Cart block with images cropped to 1:2_
<img src="https://user-images.githubusercontent.com/3616980/85136815-7d6a7000-b240-11ea-85b1-4311bad5c1cf.png" alt="Screenshot" width="670" />

### How to test the changes in this Pull Request:

1. Go to Appearance > Customize > WooCommerce > Product images and change the cropping options.
2. Test the Cart, Checkout and Review blocks (for Review blocks, you might need to change its attributes so it shows the product image instead of the customer image) and verify they show the cropped image.
3. Edit an old All Products block and verify the block didn't invalidate.
4. Edit it and select the Product image inner block. There, toggle the _Image sizing_ attribute.
5. Verify when _Cropped_ is selected, the cropped image is displayed.
6. Repeat the process with the Product block.

### Changelog

> Blocks now respect the product image cropping settings. For the All Products block, the user can switch between the cropped thumbnail and the full size image.
